src/server/lib/middleware/loggedInGuard.ts

```diff
@@ -0,0 +1,80 @@
+import { NextFunction, Request } from 'express';
+import { getSession } from '../okta/api/sessions';
+import { clearOktaCookies } from '@/server/routes/signOut';
+import { logger } from '../serverSideLogger';
+import { addQueryParamsToPath } from '@/shared/lib/queryParams';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { getConfiguration } from '../getConfiguration';
+import { read } from '../idapi/auth';
+import { IDAPIAuthStatus } from '@/shared/model/IDAPIAuth';
+import { clearIDAPICookies } from '../idapi/IDAPICookies';
+const { okta, defaultReturnUri } = getConfiguration();
+
+/** Middleware function which will redirect the client to the base Guardian
+ * domain if a valid logged in session is found. Currently checks both IDAPI and
+ * Okta session validity.
+ */
+export const loggedInGuard = async (
```

Comment:
would `blockLoggedInUsers` be more descriptive ?

Reply:
Yes, that's good!

Reply:
Or `redirectLoggedInUsers` might be even better.

Reply:
👍

Reply:
I'm going with `redirectIfLoggedIn`!